src/services/catalog.service.ts

```diff
@@ -117,6 +117,15 @@ export class CatalogService {
   }
 
   public getProjectCatalogItems(projectName: string, includeImages: boolean = true, includeTemplates: boolean = true, partialObjectMetadataList: boolean = false ) {
+    // Never fetch items from the `openshift` namespace as project items since
```

Comment:
Is this the right place for this logic?  If its `getProjectCatalogItems(projectName)`, I expect it to return the catalog items for whatever project, even if `openshift`.  Should calling code instead be wrapped in the `if` statement?

I don't see this method used elsewhere in `/src`, but in `origin-web-catalog` I find:

```
./scripts/directives/processTemplateDialog.js:      Catalog.getProjectCatalogItems(ctrl.templateProjectName, false, true).then( _.spread(function(catalogServiceItems, errorMessage) {
./scripts/directives/nav.js:                var projectItemsPromise = Catalog.getProjectCatalogItems(currentProjectName).then(_.spread(function(catalogServiceItems) {
./scripts/controllers/projectBrowseCatalog.js:          var projectItemsPromise = Catalog.getProjectCatalogItems(currentProjectName).then( _.spread(function(catalogServiceItems, errorMessage) {
```

I'm assuming this guard would be in place then for someone who is in the `openshift` namespace & happens to browse the catalog?

Reply:
Yeah, I debated this. The problem is we're adding namespaced service classes, and we *should* fetch those for the openshift namespace (just not templates and image streams). So this isn't always a no-op. The only place we can have that logic is here the way the code is structured.

My first implementation had a bug as well where I fixed it in one place and not the other. And I found myself copying code between the two (it was a bit more than just an `if` since we use `$q.all`).

`processTemplateDialog` probably shouldn't be using this method :/

> I'm assuming this guard would be in place then for someone who is in the openshift namespace & happens to browse the catalog?

Yes. Today you get a runtime error when you try to do that since there are multiple items with the same UID.

Reply:
Gotcha.  Thoughts on something like:

```JavaScript 
// a little more verbose, but only if overrdiging defaults. 
getProjectCatalogItems('my-project', {
  templates: true, 
  images: true,
  partialObjectMetadata: true, 
  openshiftNamespace: true
});
getProjectCatalogItems('my-project', {templates: false});
// vs, having to go look up what the booleans mean when using:
getProjectCatalogItems('my-project', false, false, true);
getProjectCatalogItems('my-project', false, true, false);
getProjectCatalogItems('my-project', false, false);
```
Maybe adding openshift namespace to that as a config object option, though default to `false` as well.  

I don't have super strong feelings, but it could be nice especially considering the need to look up the positional args across our repos.